### PR TITLE
fbp-generator: Check if symbol is a valid string.

### PR DESCRIPTION
**Pull request by cabelitos**
As the node symbol and the options symbol will be used in the generated C code,
it must be valid string in order to compile it in C.

Signed-off-by: Guilherme Iscaro <guilherme.iscaro@intel.com>

### CODE REVIEW

File: src/bin/sol-fbp-generator/type-store.c [hunk 1]
```diff
@@ -796,6 +817,7 @@ type_store_read_from_json(struct type_store *store, struct sol_str_slice input)
         if (!read_type(&decoder, desc)) {
             type_description_fini(desc);
             sol_vector_del_last(&store->types);
+            return false;
```
Comment (edersondisouza) [inline src/bin/sol-fbp-generator/type-store.c]: A wild `return false;` appears.
1. Delete
2. Leave it there, it was forgotten to start with!

File: src/bin/sol-fbp-generator/type-store.c [hunk 2]
```diff
@@ -699,6 +699,23 @@ read_options(struct decoder *d, struct sol_vector *options)
...
 
 static bool
+is_valid_symbol(const char *symbol)
+{
+    size_t i;
+
+    for (i = 0; symbol[i]; i++) {
+        if ((i == 0 && isdigit((int)symbol[i])) ||
+            (symbol[i] != '_' && !isalnum((int)symbol[i]))) {
+            fprintf(stderr, "Symbol '%s' with invalid character '%c' - pos: '%zu'\n",
```
Comment (barbieri) [inline src/bin/sol-fbp-generator/type-store.c]: prefix with `ERROR: `? or some other we're using in the other error cases of this file?

Comment (cabelitos) [inline src/bin/sol-fbp-generator/type-store.c]: Fair. I will add the `Error:` prefix

File: src/bin/sol-fbp-generator/type-store.c (hunk 1 above)
Comment (cabelitos) [inline src/bin/sol-fbp-generator/type-store.c]: @barbieri what you think about leaving this `return false` here?

Comment (barbieri) [inline src/bin/sol-fbp-generator/type-store.c]: ok for me, that means we'll stop early.

**Comment by cabelitos** [comment]
Merged

**Pull request closed without merging**